cli/main.c

```diff
@@ -157,33 +145,62 @@ int main(int argc, char** argv) {
   // FIXME: this is temp till arg parse implemented.
   PkResult result;
 
-  if (argc == 1) {
+  static const char* const usage[] = {
+    "pocket [option] ... [-c cmd | file | -] [arg] ...",
+    NULL,
+  };
+
+  bool debug = false, quiet = false, version = false, help = false;
+  const char** cmd = NULL;
+
+  struct argparse_option cli_opts[] = {
+    OPT_STRING('c', "cmd", &cmd, "Evaluate and run the passed string."),
+    OPT_BOOLEAN('d', "debug", &debug, "Compile and run a debug version."),
+    OPT_BOOLEAN('h', "help", &help, "Prints this help message and exit."),
+    OPT_BOOLEAN('q', "quiet", &quiet,
+               "Don't print version and copyright statement on REPL startup."),
+    OPT_BOOLEAN('v', "version", &version,
+               "Prints the pocketlang version number and exit."),
+    OPT_END(),
+  };
+
+  struct argparse argparse;
+  argparse_init(&argparse, cli_opts, usage, 0);
+  argc = argparse_parse(&argparse, argc, argv);
+
+  if (debug) {
+    options.debug = debug;
     options.repl_mode = true;
     repl(vm, &options);
-
-  } if (argc >= 3 && strcmp(argv[1], "-c") == 0) {
-
+  } else if (help) argparse_usage(&argparse);
+  else if (version) printf("pocketlang %s\n", PK_VERSION_STRING);
+  else if (cmd) {
```

Comment:
checking options one by one will prevent us from combining some arguments for an example running a file with debug options won't work.

```
pocket -d file.pk
```

Reply:
@ThakeeNathees, I'm going to actually leave this one for you to take on please. I'm a bit low brain now. Thank you!

Reply:
@xSavitar It's okay we're here to learn, You really did well with this PR. Once you done other review, I'll have a final review and merge it. And do it by myself.

Reply:
I honestly appreciate. I've learnt a lot here to be honest. I'm still trying to understand the code base bit by bit, so please I know very little here, treat me like a baby but don't pet me :)